_delphi_utils_python/delphi_utils/geomap.py

```diff
@@ -580,3 +595,16 @@ def get_geo_values(self, geo_type):
             crosswalk = pd.read_csv(stream, dtype=str)
             self.geo_lists[geo_type] = set(crosswalk[geo_type])
             return self.geo_lists[geo_type]
+
+    def _megafips_to_zero(self, fips_list):
+        output = set()
+        fips_set = set(fips_list)
+        state_fips_codes = {str(x).zfill(2) + "000" for x in range(1, 73)}
+        for state in state_fips_codes:
+            is_present = state in fips_set
+            # check if megafips is the only code reported for that state
+            is_only_fips = not any(i.startswith(state[:2]) and
+                                   not i.endswith(state[2:]) for i in fips_set)
+            if is_present and not is_only_fips:
+                output.add(state)
```

Comment:
uh this appears to be O(73n)

would it make more sense to do something like
```
index = set()
for fips in fips_list:
    if fips[2:] != '000':
        index.add(fips[:2])
return {statecode + '000' for statecode in index}
```

Reply:
ah right, since we just create the mask by checking inclusion into this set we don't actually need to filter the values to whats present in the df. good idea.